content/doc/developer/testing/index.adoc

```diff
@@ -406,8 +365,53 @@ Search is performed in the order of the brackets at the line start. The fallback
 The choice of zip and directory depends on the nature of the test data, as well as the size of it.
 
 === Configuring an SCM
-TODO: Write this section.
-You can create a Git repository during a test using `@GitSampleRepoRule`.
+
+Projects typically check out code from an SCM before running the build steps, and the test harness provides a few dummy SCM implementations which make it easy to "check out" files into the workspace.
+
+==== Using a Dummy SCM
+
+The simplest of these is the link:https://javadoc.jenkins.io/component/jenkins-test-harness/?org/jvnet/hudson/test/SingleFileSCM.html[`SingleFileSCM`] which, as its name suggests, provides a single file during checkout.
+For example:
+
+[source,java]
+----
+@Rule public JenkinsRule j = new JenkinsRule();
+
+@Test public void customizeWorkspaceWithFile() throws Exception {
+  // Create a Freestyle project with a dummy SCM
+  FreeStyleProject project = j.createFreeStyleProject();
+  project.setScm(new SingleFileSCM("greeting.txt", "hello"));
+  // …
+}
+----
+Once a build of this project starts, the file `greetings.txt` with the contents `hello` will be added to the workspace during the SCM checkout phase.
+
+There are additional variants of the `SingleFileSCM` constructor which let you create the file contents from a byte array, or by reading a file from the resources folder, or another `URL` source.
+For example:
+
+[source,java]
+----
+import io.jenkins.myplugin;
+
+// Reads the contents from `src/test/resources/io/jenkins/myplugin/test.json`
+project.setScm(new SingleFileSCM("data.json", getClass().getResource("test.json")));
+
+// Reads the contents from `src/test/resources/test.json` — note the slash prefix
+project.setScm(new SingleFileSCM("data.json", getClass().getResource("/test.json")));
+----
+
+If you want to provide more than a single file, you can use link:https://javadoc.jenkins.io/component/jenkins-test-harness/?org/jvnet/hudson/test/ExtractResourceSCM.html[`ExtractResourceSCM`], which will extract the contents of a given zip file into the workspace:
+
+[source,java]
+----
+import io.jenkins.myplugin;
+
+// Extracts `src/test/resources/io/jenkins/myplugin/files-and-folders.zip` into the workspace
+project.setScm(new ExtractResourceSCM(getClass().getResource("files-and-folders.zip")));
+----
+
+==== Using a Git SCM
+You can create a Git repository during a test using `@GitSampleRepoRule`. This can be found in the link:https://github.com/jenkinsci/git-plugin/blob/master/src/test/java/jenkins/plugins/git/GitSampleRepoRule.java[git-plugin].
```

Comment:
```suggestion
You can create a Git repository during a test using `@GitSampleRepoRule`.
This can be found in the link:https://github.com/jenkinsci/git-plugin/blob/master/src/test/java/jenkins/plugins/git/GitSampleRepoRule.java[git-plugin].
```
Small formatting adjustment to make it one sentence per line.